autofs-5.1.9 - fix get parent multi-mount check in try_remount()

From: Ian Kent <raven@themaw.net>

In commit 635b90ecc (autofs-5.1.8 - fix mount tree startup reconnect)
when getting the parent the check for a multi-mount should include a
check for the root of the multi-mount as well since the root does not
set the parent.

We could set the root parent to itself but that may have side effects
because the convention has always been the parent is NULL for the
multi-mount root.

Reported-by: Roberto Bergantinos Corpas <rbergant@redhat.com>
Suggested-by: Roberto Bergantinos Corpas <rbergant@redhat.com>

Signed-off-by: Ian Kent <raven@themaw.net>
---
 CHANGELOG    |    1 +
 lib/mounts.c |    2 +-
 2 files changed, 2 insertions(+), 1 deletion(-)

--- autofs-5.1.7.orig/CHANGELOG
+++ autofs-5.1.7/CHANGELOG
@@ -159,6 +159,7 @@
 - fix expire retry looping.
 - allow -null map in indirect maps.
 - fix multi-mount check.
+- fix get parent multi-mount check in try_remount().
 
 25/01/2021 autofs-5.1.7
 - make bind mounts propagation slave by default.
--- autofs-5.1.7.orig/lib/mounts.c
+++ autofs-5.1.7/lib/mounts.c
@@ -2878,7 +2878,7 @@ int try_remount(struct autofs_point *ap,
 	}
 
 	me->flags &= ~MOUNT_FLAG_DIR_CREATED;
-	mapent = IS_MM(me) ? MM_PARENT(me) : me;
+	mapent = IS_MM(me) && !IS_MM_ROOT(me) ? MM_PARENT(me) : me;
 	/* Direct or offset mount, key is full path */
 	if (mapent->key[0] == '/') {
 		if (!is_mounted(mapent->key, MNTS_REAL))
